### PR TITLE
Documentation URL 

### LINKED ISSUE #14947: Util.php does not use correct MySql version if version is 8.0 or above
If Mysql version is 8.0 or above, phpmyadmin/libraries/classes/Util.php still uses max version 5.7

See Util.php starting from line 358:
`           if ($serverVersion >= 50700) {`
`                $mysql = '5.7';`
`            } elseif ($serverVersion >= 50600) {`
`                $mysql = '5.6';`
`            } elseif ($serverVersion >= 50500) {`
`                $mysql = '5.5';`
`            }`

It should be
`            if ($serverVersion >= 80000) {`
`                $mysql = '8.0';`
`            } elseif ($serverVersion >= 50700) {`
`                $mysql = '5.7';`
`            } elseif ($serverVersion >= 50600) {`
`                $mysql = '5.6';`
`            } elseif ($serverVersion >= 50500) {`
`                $mysql = '5.5';`
`            }
`

### LINKED ISSUE #15790: MySQL Documentation button
It's tooltip is just `Documentation` rather than `MySQL Documentation`.
It links to 5.7 rather than 8.0.
It links to MySQL rather then MariaDB on a MariaDB server.

(nav panel)

### PR DESCRIPTION
**Pull request by victorphoenix3**
Fixes #14947
Fixes #15790

### CODE REVIEW

**Review by williamdes — APPROVED** [review]

**Comment by victorphoenix3** [comment]
@williamdes  is this fine or should I move the code to  https://github.com/williamdes/mariadb-mysql-kbs ?

File: libraries/classes/Util.php [hunk 1]
```diff
@@ -245,6 +247,24 @@ public static function getMySQLDocuURL($link, $anchor = '')
...
 
+    /**
+     * Get a URL link to the official documentation page of either MySQL
+     * or MariaDB depending on the databse server
+     * of the user.
+     * @param bool $isMariaDB if the database server is MariaDB
+     *
+     * @return string The URL link
+     */
+    public static function getDocuURL(bool $isMariaDB = false)
```
Comment (williamdes) [inline libraries/classes/Util.php]:
```suggestion
    public static function getDocuURL(bool $isMariaDB = false): string
```

**Review by williamdes — APPROVED** [review]
For me it is okay, no need for now to move the links but maybe in a near future so they can receive updates and version handling

File: libraries/classes/Util.php (hunk 1 above)
Comment (victorphoenix3) [reply libraries/classes/Util.php]: updated

File: libraries/classes/Util.php [hunk 2]
```diff
@@ -245,6 +247,24 @@ public static function getMySQLDocuURL($link, $anchor = '')
...
+     *
+     * @return string The URL link
+     */
+    public static function getDocuURL(bool $isMariaDB = false): string
+    {
+        if ($isMariaDB) {
+            $url = 'https://mariadb.com/kb/en/documentation/';
+            return Core::linkURL($url);
+        } else {
+            $this->getMySQLDocuURL('');
```
Comment (williamdes) [inline libraries/classes/Util.php]: Is it missing a return?

Comment (victorphoenix3) [reply libraries/classes/Util.php]: corrected it.

File: libraries/classes/Util.php [hunk 3]
```diff
@@ -245,6 +247,24 @@ public static function getMySQLDocuURL($link, $anchor = '')
...
+     * @param bool $isMariaDB if the database server is MariaDB
+     *
+     * @return string The URL link
+     */
+    public static function getDocuURL(bool $isMariaDB = false): string
+    {
+        if ($isMariaDB) {
+            $url = Core::linkURL('https://mariadb.com/kb/en/documentation/');
+        } else {
+            $url = $this->getMySQLDocuURL('');
```
Comment (williamdes) [inline libraries/classes/Util.php]: I would prefer to have 2 return statements
Also please check Travis CI you will see that phpstan pointed out that you can not use $this in a static method

:)

Comment (victorphoenix3) [reply libraries/classes/Util.php]: fixed :)

File: libraries/classes/Util.php [hunk 4]
```diff
@@ -245,6 +247,25 @@ public static function getMySQLDocuURL($link, $anchor = '')
...
+     * @return string The URL link
+     */
+    public static function getDocuURL(bool $isMariaDB = false): string
+    {
+        if ($isMariaDB) {
+            $url = 'https://mariadb.com/kb/en/documentation/';
+            return Core::linkURL($url);
+        } else {
+            $url = self::getMySQLDocuURL('');
+            return $url;
```
Comment (williamdes) [inline libraries/classes/Util.php]: Why can you not return the function directly?

Comment (williamdes) [reply libraries/classes/Util.php]: Just that it would be more readable :)

**Review by williamdes — APPROVED** [review]

File: libraries/classes/Util.php (hunk 4 above)
Comment (victorphoenix3) [reply libraries/classes/Util.php]: done

File: libraries/classes/Util.php [hunk 5]
```diff
@@ -245,6 +247,24 @@ public static function getMySQLDocuURL($link, $anchor = '')
...
+     * @return string The URL link
+     */
+    public static function getDocuURL(bool $isMariaDB = false): string
+    {
+        if ($isMariaDB) {
+            $url = 'https://mariadb.com/kb/en/documentation/';
+            return Core::linkURL($url);
+        } else {
+            return self::getMySQLDocuURL('');
+        }
```
Comment (mauriciofauth) [inline libraries/classes/Util.php]: The `else` statement is useless here. You can remove it.

**Review by MauricioFauth — CHANGES_REQUESTED** [review]

**Review by MauricioFauth — APPROVED** [review]

**Comment by williamdes** [comment]
Thank you @victorphoenix3 for this enhancement !

**Pull request merged**